tests/generated/YGAspectRatioTest.cpp

```diff
@@ -80,28 +80,28 @@ TEST(YogaTest, aspect_ratio_does_not_stretch_cross_axis_dim) {
 
   ASSERT_FLOAT_EQ(0, YGNodeLayoutGetLeft(root_child0_child0));
   ASSERT_FLOAT_EQ(0, YGNodeLayoutGetTop(root_child0_child0));
-  ASSERT_FLOAT_EQ(300, YGNodeLayoutGetWidth(root_child0_child0));
-  ASSERT_FLOAT_EQ(197, YGNodeLayoutGetHeight(root_child0_child0));
+  ASSERT_FLOAT_EQ(285, YGNodeLayoutGetWidth(root_child0_child0));
+  ASSERT_FLOAT_EQ(187, YGNodeLayoutGetHeight(root_child0_child0));
```

Comment:
> Regenerated tests (as some aspect ratio tests seem to be out of date compared to the fixtures)

@rozele added this specific fixture yesterday with https://github.com/facebook/yoga/commit/50ecd981414b85e786ffec8110154dd2caf82a9e.

It it disabled right now regardless, but I am curious where the difference comes from. @rozele @nicoburns do you know what Chrome versions/platforms created these results?

Reply:
I'm getting these results on Chrome 117 and 118 (just updated) on  macOS (arm64)

Reply:
> I'm getting these results on Chrome 117 and 118 (just updated) on macOS (arm64)

I just pulled down this PR and re-ran gentest, it's reverting these changed.

Chrome - Version 118.0.5993.70 (Official Build) (arm64)

Reply:
I get the same as the changes this PR makes (arm64 macOS 117)... Something's odd here.

Reply:
Maybe chrome running an A/B test?

Reply:
So I pasted the test HTML into the template and opened it up in Chrome, and I think I know what's going here. It's because there is a div with `overflow: scroll`, and I have set my OS to "always show scrollbars", whereas @rozele, I'm assuming you have it set to one of the settings where scrollbars don't take up space. If I re-run the gentests with scrollbars set to overlay scrollbars then I get the same results as comitted.

Reply:
Interestingly, the *reason* I have my OS set to "always show scrollbars" is because:

- Taffy now takes the space taken up by scrollbars when `overflow: scroll` is set
- The only way to test that properly using gentests is to ensure that scrollbars are taking up space when you generate the tests
- So Taffy's gentest runner now has an assert to check for that and refuses to run if you don't have space-taking scrollbars enabled.

I'm guessing Yoga doesn't reserve space for scrollbars? (which kinda makes sense given that it's mobile focussed - although I imagine it could cause issues for react-native-windows)

Want to me to remove that commit from this PR?

Reply:
Hmm, maybe that's why `overflow: scroll` wasn't supported in gentest before my diff 😅

I'd hate to say we shouldn't use `overflow: scroll` in fixtures arbitrarily, especially since they can certainly be a source of bugs, but maybe having fixtures for them is a bad idea (unless there's a way we can make gentest detect the scrollbar setting and warn / switch to the "correct" default before running)

Reply:
Feel free to keep it in. We can try to figure out the right solution for this after (I think we might be able to avoid the ambiguity in the fixture).

On reserving space for scrollbar on non-mobile, I think WinUI default has shy scrollbars that don't take up space until hovered over, and then they go over content, https://learn.microsoft.com/en-us/windows/apps/design/controls/scroll-controls

I think in the other cases, Scrollbar being present shrinks the available space of the layout, but isn't handled explicitly by Yoga.

Reply:
Edit: looks like there is a `--hide-scrollbars` Chrome option we can pass to Chromedriver, to make this deterministic, and match existing behavior.

Reply:
> unless there's a way we can make gentest detect the scrollbar setting and warn / switch to the "correct" default before running

This is definitely possible (you can test for the setting in the browser by creating a node with `overflow: scroll` set and checking for it's width), but it might require a bit of work to thread the error message back through to the ruby test runner script so that it could be displayed nicely to the user.

I definitely think it would be good to this kind of error reporting improvements at some point, as currently the gentest script has a nasty habit of just silently crashing (sometimes having written half the tests!) if anything goes wrong (like an unexpected property value).

Reply:
> Edit: looks like there is a `--hide-scrollbars` Chrome option we can pass to Chromedriver, to make this deterministic, and match existing behavior. I tacked this onto the imported PR for now.

I guess that works. I think I dismissed that for Taffy because I wanted the opposite: "force show scrollbars", and that doesn't seem to exist.